transcoders.go

```diff
@@ -294,11 +295,15 @@ var TranscoderP2P = NewTranscoderFromFunctions(p2pStB, p2pBtS, p2pVal)
 
 func p2pStB(s string) ([]byte, error) {
 	// the address is a varint prefixed multihash string representation
-	m, err := mh.FromB58String(s)
+	c, err := cid.Decode(s)
```

Comment:
This won't work for ed25519 keys.

Reply:
Lol, well I should've gotten that. 🤦

Reply:
Note: I think you can also check "does it start with 1 or Q". If so, decode from base58.

Reply:
Sure, I can just copy the code from https://github.com/libp2p/go-libp2p-core/blob/b77fd280f2bfcce22f10a000e8e1d9ec53c47049/peer/peer.go#L172, but that's technically a spec violation.